workload/metrics.go

```diff
@@ -9,26 +12,39 @@ var (
 			Name: "operations_total",
 			Help: "How many user operations are attempted, partitioned by operation.",
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 	opsFailed = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Name: "operations_failed_total",
 			Help: "How many user operations failed, partitioned by operation.",
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 	opDuration = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Name:    "operation_duration_milliseconds",
 			Help:    "Duration of user operations in milliseconds, partitioned by operation.",
 			Buckets: []float64{25.0, 50.0, 100.0, 250.0, 500.0, 1500.0, 2500.0},
 		},
-		[]string{"operation"},
+		[]string{"operation", "state"},
 	)
 
 	// Maps from the operation to an attempted/failed metric labelled with the operation
-	attemptMetrics  = map[string]prometheus.Counter{}
-	failedMetrics   = map[string]prometheus.Counter{}
-	durationMetrics = map[string]prometheus.Observer{}
+	attemptMetrics  = map[string]map[string]prometheus.Counter{}
+	failedMetrics   = map[string]map[string]prometheus.Counter{}
+	durationMetrics = map[string]map[string]prometheus.Observer{}
+
+	States   = []string{"warmup", "steady", "warm_down"}
+	warmTime = time.Minute
 )
+
+func MetricState(start time.Time, end time.Time) string {
+	state := "steady"
+	if time.Now().Sub(start) < warmTime {
+		state = "warmup"
```

Comment:
If there will never be other states, consider making the array fixed size and grabbing it directly from the string array which is sort of a structure in this case?  That makes me think we're just abusing string array here.  I'm probably being a bit pedantic, but when I see states, I kind of want some structure around them to avoid misuse.  For example, someone could type "warm_down" when it's "ramp down" and then the program will be broken in confusing ways.  :)

Looks like it's better to use a const/enum in golang:
https://stackoverflow.com/questions/57327764/representing-states-in-go

it's a little more typing, but then you'll have things like `state = WarmingUp`, though I wonder if something like opPhase might be better since the word state is just so generic.